src/framework/chart/LineChart.jsx

```diff
@@ -163,6 +168,66 @@ export default class LineChart extends Component {
     // Transition each line from the previous shape to the new shape.
     lines.transition().duration(duration)
       .attr('d', item => lineGenerator(item.values));
+
+    // TOOLTIP /////////////////////////////////////////////////////////////////
```

Comment:
This marker indicates that probably the following code deserve to be a utility function or something for better organisation of the code.